File: discovery/src/main/scala/com/socialthingy/plusf/p2p/Callbacks.scala
package com.socialthingy.plusf.p2p

trait Callbacks {
  def data(content: Any): Unit

  def discoveryTimeout(): Unit
  def discoveryCancelled(): Unit
  def waitingForPeer(): Unit
  def connectedToPeer(port: Int): Unit
  def discovering(): Unit
  def initialising(): Unit
  def closed(): Unit
}

File: plus-f/src/main/kotlin/com/socialthingy/p2p/Callbacks.kt
package com.socialthingy.p2p

import java.net.InetSocketAddress

interface Callbacks {
    fun data(content: Any)
    fun discoveryTimeout()
    fun discoveryCancelled()
    fun waitingForPeer()
    fun connectedToPeer(address: InetSocketAddress)
    fun discovering()
    fun initialising()
    fun closed()
}
